### DISCUSSION TITLE
Adult Website Detection

### DISCUSSION THREAD

**Discussion by BradKML** [ActivityWatch/activitywatch#591] [Ideas]
Considering monitoring activity is important, adding these into a separate category might be useful:
- https://github.com/StevenBlack/hosts
- https://github.com/Import-External-Sources/pornhosts
- https://github.com/4skinSkywalker/Anti-Porn-HOSTS-File
- https://github.com/mhxion/pornaway
- https://github.com/mrvivacious/PorNo-_Porn_Blocker

**Comment by prplecake** [comment]
I'm not sure if this is a recently added feature or not, but you can in v0.11.

For the category RegEx just put the name of the website.

**Reply by BradKML** [reply]
Most of these are domain name identifiers, would RegEx be needed?

**Reply by prplecake** [reply]
To define the category in ActivityWatch, yes.

**Comment by RESDXChgfore9hing** [comment]
is there like a generic way of adding category?
and is the category we created modular?(can be exported and imported?)

**Reply by BradKML** [reply]
That is a good idea ngl, imagine importing from uBO blocklists. Might have to ask @prplecake for a function that convert HOST files into a list of URLs with the RegEx `0\.0\.0\.0 ([^\n]+)\n`
